python/iceberg/api/types/conversions.py

```diff
@@ -81,11 +82,11 @@ def from_partition_string(type_var, as_string):
         return part_func(as_string)
 
     @staticmethod
-    def to_byte_buffer(type_var, value):
+    def to_byte_buffer(type_id, value):
         try:
-            return Conversions.to_byte_buff_mapping.get(type_var.type_id)(type_var, value)
+            return Conversions.to_byte_buff_mapping.get(type_id)(type_id, value)
         except KeyError:
-            raise RuntimeError("Cannot Serialize Type: %s" % type_var)
+            raise RuntimeError("Cannot Serialize Type: %s" % type_id)
```

Comment:
Since you're touching this code anyway, it may be more appropriate to raise either TypeError or NotImplementedError here.  There are a number of exceptions in this codebase that raise Runtime but should raise something more specific.

Reply:
Yep, update accordingly.